mypy/plugins/attrs.py

```diff
@@ -290,6 +292,12 @@ def attr_class_maker_callback(ctx: 'mypy.plugin.ClassDefContext',
         if kw_only:
             ctx.api.fail(KW_ONLY_PYTHON_2_UNSUPPORTED, ctx.reason)
             return
+    if ctx.api.options.python_version[:2] < (3, 10):
+        if match_args:
+            ctx.api.fail("match_args is not supported before Python 3.10", info.defn)
```

Comment:
Is this what attrs does at runtime? It's a bit strange to me because it seems harmless to set it before 3.10. For example, you might use it in a library if you want 3.10 users to be able to use `match`, but still support older versions too.

Reply:
Yeah looks like we set it conditionally, presumably to avoid a little work and since it'd be useless.

Since the check is in runtime and based on the running interpreter, I'm not sure I understand your concern. Can you elaborate? We can relax this if needed.

Reply:
The scenario I'm thinking of is:

- I'm writing a library that supports both 3.9 and 3.10
- The library exposes an attrs class in its API
- I want 3.10 users to be able to use the class with `match`

In that case, I'd want to be able to just set `match_args=True`.

It looks like that's already how the runtime works (it just doesn't set the attribute on 3.9 or lower), but the code in this PR would make mypy emit an error if it's checking in 3.9 mode and sees `match_args=True`.

Reply:
Ah, right, I see. It's true that the **parameter** exists on earlier Pythons, so it's not an error to use it there. And you can't use `__match_args__` on earlier Pythons anyway (I guess you could manually do something with it, but who does that).

Also, `match_args` defaults to True.

So this check can essentially be removed?

Reply:
Addressed! 👍